flaml/automl/model.py

```diff
@@ -1222,21 +1226,26 @@ def predict(self, X, **pred_kwargs):
 
         new_trainer = self._init_model_for_predict()
 
-        if self._task not in NLG_TASKS:
-            predictions = new_trainer.predict(test_dataset)
-        else:
-            predictions = new_trainer.predict(
-                test_dataset,
-                metric_key_prefix="predict",
+        try:
+            if self._task not in NLG_TASKS:
+                predictions = new_trainer.predict(test_dataset)
+            else:
+                predictions = new_trainer.predict(
+                    test_dataset,
+                    metric_key_prefix="predict",
+                )
+
+            post_y_pred, _ = postprocess_prediction_and_true(
+                task=self._task,
+                y_pred=predictions.predictions,
+                tokenizer=self.tokenizer,
+                hf_args=self._training_args,
+                X=X,
             )
-        post_y_pred, _ = postprocess_prediction_and_true(
-            task=self._task,
-            y_pred=predictions.predictions,
-            tokenizer=self.tokenizer,
-            hf_args=self._training_args,
-            X=X,
-        )
-        return post_y_pred
+            return post_y_pred
+        except ZeroDivisionError:
+            logger.warning("Zero division error appeared in HuggingFace Transformers.")
+            return np.array([0] * len(test_dataset))

```

Comment:
Isn't it just the `predict()` calls that could raise the exception? Perhaps it would be better to create a `kwargs` variable that will be either empty (`{}`) or `{"metric_key_prefix": "predict"}`, and then call once with a shorter `try-except` clause? 
AFAIK, it is not good practice to include this much code within a `try-except` clause.

Reply:
Thanks for the comments; addressed.